src/devices/hideki.c

```diff
@@ -37,31 +47,54 @@ static int hideki_ts04_callback(bitbuffer_t *bitbuffer) {
         // check parity
         uint8_t parity = ((b[i+offset+1] >> (7 - i%8)) ^ 0xFF) & 0x01;
         if(parity != byteParity(packet[i]))
-            return 0;
```

Comment:
To handle the different data length of the different types of sensors. But I agree that the code should return immediately if parity error occurs and the datalength differs from every type of sensor. Will fix in next pull request.